.github/scripts/build.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -eux
+
+. .github/scripts/env.sh
+. .github/scripts/common.sh
+
+uname -a
+uname -p
+uname
+pwd
+env
+
+# ensure ghcup
+if ! command -v ghcup ; then
+	install_ghcup
+fi
+
+# ensure cabal-cache
+download_cabal_cache "$HOME/.local/bin/cabal-cache"
+
+
+# build
+ecabal update
+ghcup install ghc "${GHC_VERSION}"
+ghcup set ghc "${GHC_VERSION}"
+"ghc-${GHC_VERSION}" --info
+"ghc" --info
+
+mkdir -p "$CI_PROJECT_DIR/out/${ARTIFACT}"
+mkdir -p "$CI_PROJECT_DIR/out/plan.json"
+
+case "$(uname)" in
+    MSYS_*|MINGW*)
+		args=( -O2 -w "ghc-$GHC_VERSION" --project-file cabal.project --disable-profiling --disable-tests --enable-executable-stripping ${ADD_CABAL_ARGS})
+
+		# Shorten binary names
+		# due to MAX_PATH issues on windows
+		sed -i.bak -e 's/haskell-language-server/hls/g' \
+			   -e 's/haskell_language_server/hls/g' \
+			   haskell-language-server.cabal cabal.project
+		sed -i.bak -e 's/Paths_haskell_language_server/Paths_hls/g' \
+			   src/**/*.hs exe/*.hs
+
+		# shellcheck disable=SC2068
+		build_with_cache ${args[@]} exe:hls exe:hls-wrapper
+		cp dist-newstyle/cache/plan.json "$CI_PROJECT_DIR/out/plan.json/${ARTIFACT}-ghc-${GHC_VERSION}-plan.json"
+
+		# shellcheck disable=SC2068
+		cp "$(cabal list-bin -v0 ${args[@]} exe:hls)" "$CI_PROJECT_DIR/out/${ARTIFACT}/haskell-language-server-${GHC_VERSION}${ext}"
+		# shellcheck disable=SC2068
+		cp "$(cabal list-bin -v0 ${args[@]} exe:hls-wrapper)" "$CI_PROJECT_DIR/out/${ARTIFACT}/haskell-language-server-wrapper${ext}"
+        ;;
+	*)
+		sed -i.bak -e '/DELETE MARKER FOR CI/,/END DELETE/d' cabal.project # see comment in cabal.project
```

Comment:
?

Reply:
I have no idea. This is ported from existing CI.